src/test_fixture/sharing.rs

```diff
@@ -68,19 +83,52 @@ where
     ]
 }
 
-/// Shares `input` into 3 maliciously secure replicated secret shares using the provided `rng` implementation
-///
-#[allow(clippy::missing_panics_doc)]
-pub fn share_malicious<F: Field, R: RngCore>(x: F, r: F, rng: &mut R) -> [MaliciousReplicated<F>; 3]
+/// For upgrading various shapes of replicated share to malicious.
+#[async_trait]
+pub trait IntoMalicious<F: Field, M> {
+    async fn upgrade(self, ctx: MaliciousContext<'_, F>) -> M;
+}
+
+#[async_trait]
+impl<F: Field> IntoMalicious<F, MaliciousReplicated<F>> for Replicated<F> {
+    async fn upgrade<'a>(self, ctx: MaliciousContext<'a, F>) -> MaliciousReplicated<F> {
+        ctx.upgrade(RecordId::from(0_u32), self).await.unwrap()
```

Comment:
Why do we need this? And why does this work with a hard-coded record_id of zero? If this is just calling the `upgrade` function on the context, can't the caller just do that instead?

Reply:
The idea is that different tests will use different argument shapes. A multiply will take a two-tuple; the carry protocol takes an array of "bits", some of the protocols use a two-tuple of arrays of bits.

There are two translations that we use in the tests:

1. Convert from raw values into shares.  
2. Convert from a replicated share (or collection thereof) into malicious shares.

These both need an implementation for each shape of argument.  I've defined a trait for both.  Hopefully, we won't need too many different shapes in our tests, because these are a little fiddly to implement.  But once you implement one shape, they tend to be reusable for other tests.